beets/autotag/__init__.py

```diff
@@ -14,36 +14,37 @@
 
 """Facilities for automatically determining files' correct metadata."""
 
-from collections.abc import Mapping, Sequence
-from typing import Union
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Union
 
 from beets import config, logging
-from beets.library import Album, Item, LibModel
+from beets.util import get_most_common_tags as current_metadata
```

Comment:
A bit of a nit, it would be better to expose both variants and add a deprecation warning to the `current_metadata` one if want to do it right :upside_down_face: Tbh tho it is not really too big of an issue.

Reply:
The new function is now meant to be imported directly from `beets.util` instead of `beets.autotag`. 

Actually, one of the main motivations for this refactor was to fix the `Distance` circular import issue by moving it to a separate module and _importing it directly_ from there.

But then wild _backwards compatibility_ appeared and threw my masterplan into the bin 🤣

Reply:
I _did_ update the imports though

```
$ rg 'import.*Distance'
beetsplug/chroma.py
27:from beets.autotag.distance import Distance

beets/plugins.py
57:    from beets.autotag.distance import Distance
228:        from beets.autotag.distance import Distance
241:        from beets.autotag.distance import Distance
462:    from beets.autotag.distance import Distance
476:    from beets.autotag.distance import Distance
664:    from beets.autotag.distance import Distance

beets/autotag/__init__.py
27:from .distance import Distance

beets/autotag/hooks.py
26:    from .distance import Distance
```

Reply:
In this case I'm a bit conservative because GitHub search suggested this change would break a pair of plugins.

Ultimately, this circular import is not causing much harm. I will have more changes coming related to `autotag`, so I think we will revisit this soon!